### DISCUSSION TITLE
allow "with" to be used with regular classes and structs

### DISCUSSION THREAD

**Discussion by Lavshyak** [dotnet/csharplang#7752] [Language Ideas] [answered]
```c#
//mySuperModel - object of class or struct
//SomeValue - public for setting property or field
```

now:

```c#
if(...)
{
    mySuperModel.SomeValue = 50;
    return mySuperModel; 
    //mySuperModel (with SomeValue == 50) will returns
}
```

after:

```c#
if(...) return mySuperModel with { SomeValue = 50 }; 
//mySuperModel (with SomeValue == 50) will returns.
//saving up to four lines of code
```

**Comment by DavidArno** [comment, marked as answer]
See #162. There seemed willingness within the language team to extend "withers" to work with non-record types five years ago. Unfortunately, there's not been any obvious signs of progress since as all that's really happened is the feature has been repeatedly pushed back in terms of release target and now just sits on the backlog. So this one resides in the "it might happen, one day" category.

Making some noise on that other thread _might_ help get things moving.
